src/game/TileEngine/SmokeEffects.h

```diff
@@ -34,15 +25,22 @@ struct SMOKEEFFECT
 };
 
 
-// Returns NO_SMOKE_EFFECT if none there...
-SmokeEffectKind GetSmokeEffectOnTile(INT16 sGridNo, INT8 bLevel);
+// Returns the smoke effect id for a specific tile on the current map or SmokeEffectID::NOTHING if no smoke effect is present.
+SmokeEffectID GetSmokeEffectOnTile(INT16 sGridNo, INT8 bLevel);
+// Returns the smoke effect id for the world flags of a tile or SmokeEffectID::NOTHING if no smoke effect is present.
+SmokeEffectID FromWorldFlagsToSmokeType(UINT8 ubWorldFlags);
+
+struct SOLDIERTYPE;
```

Comment:
No need for this line because JA2Types.h gets included. SOLDIERTYPE* is already used above in struct SMOKEEFFECT.

Reply:
Fixed.